samples/basic/src/free_buffer_early.c

```diff
@@ -5,6 +5,7 @@
 
 #include "wasm_export.h"
 #include "bh_read_file.h"
+#include "bh_platform.h"
```

Comment:
is this related to the rest of changes in this PR?

Reply:
If trying to compile samples/basic on Windows(mentioned in #4280) after patching `COMPILING_WASM_RUNTIME_API`, this modification can fix an error caused by `optarg` and `getopt()`.

Reply:
why bh_platfrom.h, not bh_getopt.h as nearby files do?